src/js/components/receiving/ReceivingPage.jsx

```diff
@@ -91,11 +96,27 @@ class ReceivingPage extends Component {
 
       this.save(payload, this.nextPage);
     } else {
-      this.save({ ...formValues, receiptStatus: 'COMPLETED' }, () => {
-        this.setState({ completed: true });
-        const { requisition } = formValues;
-        window.location = `/openboxes/stockMovement/show/${requisition}`;
+      let isBinLocationChosen = true;
+      _.forEach(formValues.containers, (container) => {
+        _.forEach(container.shipmentItems, (shipmentItem) => {
+          if (_.isNull(shipmentItem.binLocation.id) && this.props.hasBinLocationSupport && !(formValues.shipmentStatus === 'RECEIVED')) {
```

Comment:
you don't need to test the hasBinLocationSupport and shipmentStatus multiple times inside the loop